sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosTable.scala

```diff
@@ -64,9 +64,7 @@ private class CosmosTable(val transforms: Array[Transform],
     CosmosContainerConfig.parseCosmosContainerConfig(effectiveUserConfig, databaseName, containerName)
   private val tableName = s"com.azure.cosmos.spark.items.${clientConfig.accountName}." +
     s"${cosmosContainerConfig.database}.${cosmosContainerConfig.container}"
-  private val client = new CosmosClientBuilder().endpoint(clientConfig.endpoint)
-    .key(clientConfig.key)
-    .buildAsyncClient()
+  val client = CosmosClientCache(CosmosClientConfiguration(effectiveUserConfig, useEventualConsistency = true), None)
```

Comment:
why can't we pass the `cosmosClientStateHandle` here?

Reply:
The handle is not created, it is created after doing the point read